tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedThreadFactory/resourcedef/ManagedThreadFactoryDefinitionOnEJBServlet.java

```diff
@@ -0,0 +1,230 @@
+
+            ManagedThreadFactory threadFactory = InitialContext.doLookup("java:app/concurrent/EJBThreadFactoryA");
+
+            IntContext.set(162);
+            StringContext.set("testManagedThreadFactoryDefinitionAllAttributesEJB-2");
+
+            Thread thread1 = threadFactory.newThread(() -> {});
+            assertEquals(thread1.getPriority(), 4,
+                         "New threads must be created with the priority that is specified on " +
+                         "ManagedThreadFactoryDefinition");
+
+            BlockingQueue<Object> results = new LinkedBlockingQueue<Object>();
+
+            threadFactory.newThread(() -> {
+                results.add(Thread.currentThread().getPriority());
+                results.add(StringContext.get());
+                results.add(IntContext.get());
+                try {
+                    results.add(InitialContext.doLookup("java:app/concurrent/ContextA"));
+                } catch (Throwable x) {
+                    results.add(x);
+                }
+            }).start();
+
+            assertEquals(results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), Integer.valueOf(4),
+                         "ManagedThreadFactory must start threads with the configured priority.");
+
+            assertEquals(results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), "",
+                         "Third-party context type StringContext must be cleared from thread " +
+                         "per ManagedThreadFactoryDefinition and ContextServiceDefinition configuration.");
+
+            assertEquals(results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS), Integer.valueOf(161),
+                         "Third-party context type IntContext must be propagated to thread " +
+                         "per ManagedThreadFactoryDefinition and ContextServiceDefinition configuration " +
+                         "based on the thread context at the time the ManagedThreadFactory was looked up.");
+
+            Object lookupResult = results.poll(MAX_WAIT_SECONDS, TimeUnit.SECONDS);
+            if (lookupResult instanceof Throwable)
+                throw new AssertionError().initCause((Throwable) lookupResult);
+            assertTrue(lookupResult instanceof ContextService,
+                       "Application context must be propagated to thread " +
+                       "per ManagedThreadFactoryDefinition and ContextServiceDefinition configuration.");
+        } finally {
+            IntContext.set(0);
+            StringContext.set("");
+        }
+    }
+
+    /**
+     * ManagedThreadFactoryDefinition defined on an EJB with minimal attributes creates threads with normal priority
+     * and uses java:comp/DefaultContextService to determine context propagation and clearing.
+     */
+    public void testManagedThreadFactoryDefinitionDefaultsEJB() throws Throwable {
+        ManagedThreadFactory threadFactory = (ManagedThreadFactory) bean.doLookup("java:comp/concurrent/EJBThreadFactoryB");
```

Comment:
It is not correct to expect a java:comp scoped resource that was defined in an EJB to be available to a servlet.  Need to change the test.

Reply:
The EJB is looking it up.  Do you still think  this is an issue?

Reply:
Sorry, my comment was not correct. I just realized you are doing `bean.doLookup` which is making the EJB perform the lookup for you. I previously misread it as `InitialContext.doLookup` which would have caused the Servlet to be performing the lookup and wouldn't have worked. You code is fine as is.